pilot/pkg/xds/ecds.go

```diff
@@ -39,29 +40,20 @@ type EcdsGenerator struct {
 var _ model.XdsResourceGenerator = &EcdsGenerator{}
 
 func ecdsNeedsPush(req *model.PushRequest, proxy *model.Proxy) bool {
-	if proxy.Type == model.Ztunnel {
-		// Not supported for ztunnel
-		return false
-	}
-	if req == nil {
-		return true
-	}
-	// If none set, we will always push
-	if len(req.ConfigsUpdated) == 0 {
-		return true
-	}
```

Comment:
I think just breaking these 3 out is maybe fine but the rest I am not sure... (see other comments)